### ISSUE TITLE
[1.8] Schema::Field.argument(...) silently ignores blocks

### ISSUE THREAD

**Issue by cjoudrey** [rmosolgo/graphql-ruby#1347]
`GraphQL::Schema::Field.argument` method appears to silently ignore blocks passed to it.

Repro code:

```ruby
require "graphql"

class Query < GraphQL::Schema::Object
  field :field, String, null: false do
    p 'Field block'
    argument :arg, String, required: false do
      p 'Argument block'
    end
  end
end

class Schema < GraphQL::Schema
  query Query
end

Schema.types
```

We should either not accept a block or accept and call it.

I would vote for the latter so that the API is consistent with `field(...) do ... end`.

**Comment by rmosolgo** [comment]
:+1: for calling the block, sounds like a mistake 😬

**Comment by cjoudrey** [comment]
Closing as this should be fixed on latest `1.8-dev`.
